aiida_vasp/io/poscar.py

```diff
@@ -116,7 +126,11 @@ def parsevasp_to_aiida(poscar):
             symbols[symbol]
         except KeyError:
             symbol = 'X'
-        result['poscar-structure'].append_atom(position=site['position'], symbols=symbol, name=specie)
+
+        site['symbol'] = symbol
```

Comment:
For these things, I suggest we check that they do not exists. Pretty sure we will never use these in parsevasp, but we do not know what happens down the road. Better to at least get notice of it, than errors, if they are modified and have some kind of future dependency downstream. Parsevasp currently returns a new dict, but that might also change.

Reply:
Are you thinking of

```
site['symbol'] = site.get('symbol', symbol)  # don't overwrite
```

or actually

```
assert 'symbol' not in site
```

i.e. should it break or silently skip?

Reply:
Not sure. I am mostly worried down the line as we do not modify anything in parsevasp. Calling get_dict again yields another dictionary. However, if we or someone else puts these keys in parsevasp at some point I think we should raise a warning. Then, one can take action and evaluate if we should care. So I would say brake. Or we make a test that checks this. The latter is possibly the better way to avoid cluttering the code. What do you think?

Reply:
I could agree to a unit test which tests

```
poscar = ...
poscar_dict = poscar.get_dict(direct=False)
result_dict = PoscarParser.parsevasp_to_aiida(poscar)['poscar-structure']
for key in poscar_dict:
    assert poscar_dict[key] == result_dict[key]
```

If you agree, I will write that test and push it to this branch.

Reply:
Did you manage to do this Rico? Thanks.

Reply:
I have added a test based on @DropD 's proposal that checks recursively, whether any key provided by parsevasp has been overridden in `io.tests.test_poscar_io`.

Reply:
Super duper Martin. Thanks.